docs/design-documents/features/storage/Configuration/CONFIGURATION.md

```diff
@@ -0,0 +1,387 @@
+        },
+        "external_size": {
+            "help": "Size in bytes of the external block device, if not specified the maximum is the default.",
+            "value": "NULL"
+        },
+        "external_base_address": {
+            "help": "If not defined the default is from address 0",
+            "value": "NULL"
+        },    
+        "mount_point": {
+            "help": "Where to mount the filesystem. Ignored if the default file system is applied.",
+            "value": "/sd"
+        },
+        "folder_path": {
+            "help": "Path for the working directory where the FileSyetemStore stores the data",
+            "value": "/kvstore"
+        }
+    }
+}
+```
+If filesystem is not set the default filesystem and block device will be applied and blockdevice, external_size and external_base_address will be ignored.
+
+### FILESYSTEM_NO_RBP
+![FILESYSTEM](./FILESYSTEM_no_rbp.jpg)
+
+The FILESYSTEM_NO_RBP configuration resembles the EXTERNAL_NO_RBP with the difference that it uses FileSysytemStore on the external flash. By default, the FileSystemStore will use the default filesystem and the default block device. This Configuration has no support for rollback protection and therefore less secure.
+in this configuration all kvstore C API calls are mapped to the SecureStore class and this class will handle the use of the external FileSystemStore.
+This configuration can be enabled by setting storage_type in storage mbed_lib.json to FILESYSTEM_NO_RBF. 
+
+Below is the FILESYSTEM configuration mbed_lib.json
+```
+{
+    "name": "filesystem_store_no_rbp",
+    "config": {
+        "filesystem": {
+            "help": "Options are default, FAT or LITTLE. If not specified default filesystem will be used",
+            "value": "NULL"
+        },
+        "blockdevice": {
+            "help": "Options are default, SPIF, DATAFASH, QSPIF or FILESYSTEM. If not set the default block device will be used",
+            "value": "NULL"
+        },
+        "external_size": {
+            "help": "Size in bytes of the external block device, if not specified the maximum is the default.",
+            "value": "NULL"
+        },
+        "external_base_address": {
+            "help": "If not defined the default is from address 0",
+            "value": "NULL"
+        },    
+        "mount_point": {
+            "help": "Where to mount the filesystem. Ignored if the default file system is applied.",
+            "value": "/sd"
+        },
+        "folder_path": {
+            "help": "Path for the working directory where the FileSyetemStore stores the data",
+            "value": "/kvstore"
+        }
+    }
+}
+```
+
+If filesystem is not set the default filesystem and block device will be applied and blockdevice, external_size and external_base_address will be ignored 
+
+### Configuration functions API.
+Base on the parameter above a different setup function will be called. Below is a list of setup function and their description.
```

Comment:
Will be called by what? Does the user have to do anything here or will this happen automatically?

Reply:
Added explanation. User must call storage_configuration().